include/sys/types.h

```diff
@@ -301,6 +301,10 @@ typedef CODE int (*main_t)(int argc, FAR char *argv[]);
 
 #endif /* __ASSEMBLY__ */
 
+/* Defines `fd_set' and the FD_* macros for `select'. */
+
+#include <sys/select.h>
+
```

Comment:
I don't believe this is the correct way to handle this.  There is a hieararchy of header files and sys/types.h is probably the lowest level.  It should not be including higher level header files.  This is just incorrect.

Reply:
The things that must be defined by including sys/types are provided by https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html This does not include the fd_set type and FD macros.  It is incorrect to put in this file.